app/main.py

```diff
@@ -1,34 +1,123 @@
+from typing import List, Tuple
+
+
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    def __init__(
+            self,
+            row: int,
+            column: int,
+            is_alive: bool = True
+    ) -> None:
+        self.is_alive = is_alive
+        self.row = row
+        self.column = column
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    def __init__(
+            self,
+            start: Tuple[int],
+            end: Tuple[int],
+            is_drowned: bool = False
+    ) -> None:
+        self.start = start
+        self.end = end
+        self.is_drowned = is_drowned
+        self.decks = [Deck(row, column) for row in
+                      range(self.start[0], self.end[0] + 1)
+                      for column in range(self.start[1], self.end[1] + 1)]
+
+    def get_deck(self, row: int, column: int) -> object:
+        for deck in self.decks:
+            if deck.row == row and deck.column == column:
+                return deck
+        return None
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+    def generate_decks(self) -> list:
```

Comment:
without this method, my code would be crashed, because i use it below
`                if location in self.field:
                    ship = self.field[location]
                    deck = ship.get_deck(location[0], location[1])
                    if not deck.is_alive and ship.is_drowned:
                        field_representation += "X"
                    elif not deck.is_alive and not ship.is_drowned:
                        field_representation += "*"
`